homeassistant/components/group/climate.py

```diff
@@ -0,0 +1,356 @@
+            )
+        ]
+    )
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Initialize Light Group config entry."""
+    registry = er.async_get(hass)
+    entities = er.async_validate_entity_ids(
+        registry, config_entry.options[CONF_ENTITIES]
+    )
+
+    async_add_entities(
+        [
+            ClimateGroup(
+                config_entry.entry_id,
+                config_entry.title,
+                entities,
+                config_entry.options[CONF_TEMPERATURE_UNIT],
+            )
+        ]
+    )
+
+
+class ClimateGroup(GroupEntity, ClimateEntity):
+    """Representation of a climate group."""
+
+    _attr_available: bool = False
+    _attr_assumed_state: bool = True
+
+    def __init__(
+        self, unique_id: str | None, name: str, entity_ids: list[str], unit: str
+    ) -> None:
+        """Initialize a climate group."""
+        self._entity_ids = entity_ids
+
+        self._attr_name = name
+        self._attr_unique_id = unique_id
+        self._attr_extra_state_attributes = {ATTR_ENTITY_ID: entity_ids}
+
+        if "c" in unit.lower():
+            self._attr_temperature_unit = TEMP_CELSIUS
+        else:
+            self._attr_temperature_unit = TEMP_FAHRENHEIT
+
+        # Set some defaults (will be overwritten on update)
+        self._attr_supported_features = 0
+        self._attr_hvac_modes = [HVACMode.OFF]
+        self._attr_hvac_mode = None
+        self._attr_hvac_action = None
+
+        self._attr_swing_modes = None
+        self._attr_swing_mode = None
+
+        self._attr_fan_modes = None
+        self._attr_fan_mode = None
+
+        self._attr_preset_modes = None
+        self._attr_preset_mode = None
+
+    async def async_added_to_hass(self) -> None:
+        """Register callbacks."""
+
+        @callback
+        def async_state_changed_listener(event: Event) -> None:
+            """Handle child updates."""
+            self.async_set_context(event.context)
+            self.async_defer_or_update_ha_state()
+
+        self.async_on_remove(
+            async_track_state_change_event(
+                self.hass, self._entity_ids, async_state_changed_listener
+            )
+        )
+
+        await super().async_added_to_hass()
+
+    @callback
+    def async_update_group_state(self) -> None:
+        """Query all members and determine the climate group state."""
+        self._attr_assumed_state = False
+
+        states = [
+            state
+            for entity_id in self._entity_ids
+            if (state := self.hass.states.get(entity_id)) is not None
+        ]
+        self._attr_assumed_state |= not states_equal(states)
+
+        # Set group as unavailable if all members are unavailable or missing
+        self._attr_available = any(state.state != STATE_UNAVAILABLE for state in states)
+
+        def _mean(*args: float) -> float:
```

Comment:
Can we use `statistics.mean` here?
https://docs.python.org/3/library/statistics.html#statistics.mean

Reply:
Done!